package/MDAnalysis/analysis/base.py

```diff
@@ -425,34 +758,86 @@ def run(self, start=None, stop=None, step=None, frames=None,
         .. versionchanged:: 2.5.0
             Add `progressbar_kwargs` parameter,
             allowing to modify description, position etc of tqdm progressbars
-        """
-        logger.info("Choosing frames to analyze")
-        # if verbose unchanged, use class default
-        verbose = getattr(self, '_verbose',
-                          False) if verbose is None else verbose
 
-        self._setup_frames(self._trajectory, start=start, stop=stop,
-                           step=step, frames=frames)
-        logger.info("Starting preparation")
-        self._prepare()
-        logger.info("Starting analysis loop over %d trajectory frames",
-                    self.n_frames)
-        if progressbar_kwargs is None:
-            progressbar_kwargs = {}
+        .. versionchanged:: 2.8.0
+            Introduced ``backend``, ``n_workers``, ``n_parts`` and
+            ``unsupported_backend`` keywords, and refactored the method logic to
+            support parallelizable execution.
+        """
+        # default to serial execution
+        backend = "serial" if backend is None else backend
+
+        progressbar_kwargs = {} if progressbar_kwargs is None else progressbar_kwargs
+        if (progressbar_kwargs or verbose) and backend != "serial":
```

Comment:
This will incorrectly fail if `backend=BackendSerial` is passed through, even though it's functionally equivalent to `backend="serial"`.

Reply:
```suggestion
        if ((progressbar_kwargs or verbose) and 
            not (backend == "serial" or 
            issubclass(backend, BackendSerial) or
            isinstance(backend, BackendSerial))):
```
This is looking a bit wordy but I think str, classes, and instances are all allowed so we need to check for all three. Is this correct @marinegor ?

Reply:
I actually do, thanks. I can't add a test now, but I've added an issue to resolve this.